src/test/scala/com/fulcrumgenomics/util/ReadStructureTest.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.testing.UnitSpec
+
+import scala.util.Try
+
+class ReadStructureTest extends UnitSpec {
+
+  "ReadStructure" should "be built from a string" in {
+    ReadStructure("1T").toString shouldBe "1T"
+    ReadStructure("1B").toString shouldBe "1B"
+    ReadStructure("1M").toString shouldBe "1M"
+    ReadStructure("1S").toString shouldBe "1S"
+    ReadStructure("101T").toString shouldBe "101T"
+    ReadStructure("123456789T").toString shouldBe "123456789T"
+    ReadStructure("10T10B10B10S10M").toString shouldBe "10T10B10B10S10M"
```

Comment:
Would be nice on this last one to validate that each of the segments created is also the right sub-type of `ReadSegment`.